src/main/java/core/basesyntax/service/FigureSupplier.java

```diff
@@ -0,0 +1,75 @@
+package core.basesyntax.service;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+
+public class FigureSupplier {
+    private final List<Supplier<Figure>> generators;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier() {
+        this.generators = new ArrayList<>();
+        this.generators.add(this::getRandomSquare);
+        this.generators.add(this::getRandomRectangle);
+        this.generators.add(this::getRandomRightTriangle);
+        this.generators.add(this::getRandomCircle);
+        this.generators.add(this::getRandomIsoscelesTrapezoid);
+
+        this.colorSupplier = new ColorSupplier();
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("White", 10);
+    }
+
+    public Figure getRandomFigure() {
+        int randomFigureType = ThreadLocalRandom.current().nextInt(generators.size());
+        return generators.get(randomFigureType).get();
+    }
+
+    private static double getRandomDouble(double bound) {
+        return ThreadLocalRandom.current().nextDouble(bound);
+    }
```

Comment:
Don't use static methods in your solution. Instead of making `getRandomDouble` static, you can create an instance of `FigureSupplier` or another appropriate class to call this method.